faiss/impl/ScalarQuantizer.cpp

```diff
@@ -955,7 +1147,45 @@ struct SimilarityL2<1> {
     }
 };
 
+#ifdef __AVX512F__
+
+template <>
+struct SimilarityL2<16> {
+    static constexpr int simdwidth = 16;
+    static constexpr MetricType metric_type = METRIC_L2;
+
+    const float *y, *yi;
+
+    explicit SimilarityL2(const float* y) : y(y) {}
+    __m512 accu16;
+
+    FAISS_ALWAYS_INLINE void begin_16() {
+        accu16 = _mm512_setzero_ps();
+        yi = y;
+    }
+
+    FAISS_ALWAYS_INLINE void add_16_components(__m512 x) {
+        __m512 yiv = _mm512_loadu_ps(yi);
+        yi += 16;
+        __m512 tmp = _mm512_sub_ps(yiv, x);
+        accu16 = _mm512_fmadd_ps(tmp, tmp, accu16);
+    }
+
+    FAISS_ALWAYS_INLINE void add_16_components_2(__m512 x, __m512 y_2) {
+        __m512 tmp = _mm512_sub_ps(y_2, x);
+        accu16 = _mm512_fmadd_ps(tmp, tmp, accu16);
+    }
+
+    FAISS_ALWAYS_INLINE float result_16() {
+        // performs better than dividing into _mm256 and adding
+        return _mm512_reduce_add_ps(accu16);
+    }
+};
+
+#endif
+
 #ifdef __AVX2__
```

Comment:
when __AVX512__ is defined, __AVX2__ is also present in the compiler macro. This leads to unused variable errors for Meta's CI. Can you change this to

```
#elif defined(__AVX2__)
```
to silence the error? This should mirror the `if avx512 elif avx2` logic in `get_distance_computer`

Reply:
Fixed.